Automattic/dserve
# Use random internal port

### Changes

Uses [`get-port`](https://www.npmjs.com/package/get-port) instead of [`portfinder`](https://www.npmjs.com/package/portfinder) to find an internal port. The main difference is `get-port` tries random ports and [holds it for a few seconds](https://www.npmjs.com/package/get-port#beware). This should prevent `dserve` trying to use the same port twice when starting two images at the same time-ish.

### How to test

1. Follow the instructions on PCYsg-stw-p2
2. Clone this branch, install deps (`yarn`) and start dserve (`yarn start`). It will clone `wp-calypso` so it may take a while
3. Go to the 1st URL specified in 28926-pb. Once the container load, go to `/log-in` and verify you get the login page. It won't be functional because it is loaded from HTTP and from localhost.
4. Repeat with the 2nd URL in 28926-pb.
5. Inspect your containers by running `docker ps`. You should see two containers running with non-sequential internal ports:

![image](https://user-images.githubusercontent.com/975703/109455716-7d6d7180-7a57-11eb-8d5c-3065483cb868.png)
